code/game/objects/items/stacks/tiles/tile_types.dm

```diff
@@ -56,13 +63,44 @@
 			return
 		. += span_notice("Those could work as a [verb] throwing weapon.")
 
+/**
+ * Place our tile on a plating, or replace it.
+ * 
+ * Arguments:
+ * * target_plating - Instance of the plating we want to place on. Replaced during sucessful executions.
+ * * user - The mob doing the placing.
+ */
+/obj/item/stack/tile/proc/place_tile(turf/open/floor/plating/target_plating, mob/user)
+	var/turf/placed_turf_path = turf_type
+	if(!ispath(placed_turf_path))
+		return
+	if(!istype(target_plating))
+		return
 
-/obj/item/stack/tile/proc/place_tile(turf/open/T)
-	if(!turf_type || !use(1))
+	if(!replace_plating)
+		if(!use(1))
+			return
+		target_plating = target_plating.PlaceOnTop(placed_turf_path, flags = CHANGETURF_INHERIT_AIR)
+		target_plating.setDir(turf_dir)
+		playsound(target_plating, 'sound/weapons/genhit.ogg', 50, TRUE)
+		return target_plating // Most executions should end here.
+
+	// If we and the target tile share the same initial baseturf and they consent, replace em.
+	if(!target_plating.allow_replacement || initial(target_plating.baseturfs) != initial(placed_turf_path.baseturfs))
+		to_chat(user, span_notice("You cannot place this tile here directly!"))
+		return
+	to_chat(user, span_notice("You begin replacing the floor with the tile..."))
+	if(!do_after(user, 3 SECONDS, target_plating))
 		return
-	var/turf/placed_turf = T.PlaceOnTop(turf_type, flags = CHANGETURF_INHERIT_AIR)
-	placed_turf.setDir(turf_dir)
-	return placed_turf
+	if(!target_plating)
```

Comment:
should really be !QDELETED(target_plating), although in theory I think do_after might check that, but I dont' remember if it checks after the last sleep

Reply:
It'd be better to check for the type here, since a turf can only rarely be "qdeleting" in the general sense, 
when one is replaced with another all of its references switch over to the new one.

Reply:
Added. Thanks folks.